src/ucp/rndv/rndv.h

```diff
@@ -95,9 +94,12 @@ ucp_rndv_rts_is_tag(const ucp_rndv_rts_hdr_t *rts_hdr)
 {
     return rts_hdr->flags & UCP_RNDV_RTS_FLAG_TAG ||
            /* Wire compatibility with old versions of RNDV protocol which uses
-            * status instead of flags, only TAG operations could be used then */
-           ucs_unlikely((rts_hdr->status == UCS_OK) ||
-                        (rts_hdr->status == UCS_ERR_CANCELED));
+            * status instead of flags for TAG operations */
+           ucs_unlikely(((ucs_status_t)rts_hdr->flags ==
+                         (ucs_status_t)UCP_RNDV_RTS_FLAG_TAG_COMPAT_VALUE_OK) ||
+                        ((ucs_status_t)rts_hdr->flags ==
+                         (ucs_status_t)
```

Comment:
why `ucs_status_t`? can we don't cast type at all?

Reply:
no, flags is `unsigned` variant of `int8` field, but `ucs_status_t` is `signed` one

Reply:
so, maybe cast it to unsigned in `UCP_RNDV_RTS_FLAG_TAG_COMPAT_VALUE_CANCELED = UCS_ERR_CANCELED,` ?

Reply:
done